server/server/src/metrics.rs

```diff
@@ -0,0 +1,215 @@
+        }
+    });
+}
+
+pub async fn start(server_state: Arc<ServerState>) -> Result<(), ServerError<MetricsError>> {
+    loop {
+        let public_keys = get_all_public_keys(&server_state).await?;
+        METRICS_STATISTICS.total_users.set(public_keys.len() as i64);
+
+        let mut total_documents = 0;
+        let mut total_bytes = 0;
+        let mut active_users = 0;
+
+        for public_key in public_keys {
+            let mut con = server_state.index_db_pool.get().await?;
+
+            let ids = get_users_owned_files(&mut con, &public_key).await?;
+            total_documents += ids.len();
+
+            let metadatas = get_user_file_metadatas(&mut con, &ids).await?;
+            let bytes = calculate_total_document_bytes(&mut con, &metadatas).await?;
+            total_bytes += bytes;
+
+            if is_user_active(&metadatas).await? {
+                active_users += 1;
+            }
+
+            tokio::time::sleep(server_state.config.metrics.duration_between_user_metrics).await;
+        }
+
+        METRICS_STATISTICS
+            .total_documents
+            .set(total_documents as i64);
+        METRICS_STATISTICS.active_users.set(active_users);
+        METRICS_STATISTICS.total_document_bytes.set(total_bytes);
+
+        tokio::time::sleep(server_state.config.metrics.duration_between_metrics_refresh).await;
+    }
+}
+
+pub async fn get_all_public_keys(
+    server_state: &Arc<ServerState>,
+) -> Result<Vec<PublicKey>, ServerError<MetricsError>> {
+    let mut con = server_state.index_db_pool.get().await?;
+
+    let mut public_keys_k: AsyncIter<String> = con.scan_match(public_key("*")).await?;
+
+    let mut public_keys_keys = HashSet::new();
+    while let Some(item) = public_keys_k.next_item().await {
+        public_keys_keys.insert(item);
+        tokio::time::sleep(
+            server_state
+                .config
+                .metrics
+                .duration_between_getting_pub_key_key_metrics,
+        )
+        .await;
+    }
+
+    let mut public_keys: Vec<PublicKey> = vec![];
+
+    for key in public_keys_keys {
+        let public_key = con.maybe_json_get(&key).await?.ok_or_else(|| {
+            internal!(
+                "Cannot retrieve public_key despite having a valid key: {:?}",
+                key
+            )
+        })?;
+
+        public_keys.push(public_key);
+        tokio::time::sleep(
+            server_state
+                .config
+                .metrics
+                .duration_between_getting_pub_key_metrics,
+        )
+        .await;
+    }
+
+    Ok(public_keys)
+}
+
+pub async fn get_users_owned_files(
+    con: &mut deadpool_redis::Connection,
+    public_key: &PublicKey,
+) -> Result<Vec<Uuid>, ServerError<MetricsError>> {
+    con.maybe_json_get(keys::owned_files(public_key))
+        .await?
+        .ok_or_else(|| {
+            internal!(
+                "Cannot retrieve owned_files despite having a valid public_key: {:?}",
```

Comment:
```suggestion
                "Cannot retrieve owned_files for public_key: {:?}",
```

hard to determine what *valid* means if this is ever a message someone gets woken up with this.